docs/includes/openapi/openapi.adoc

```diff
@@ -102,60 +108,53 @@ This excerpt shows only a few annotations for illustration. The
 so OpenAPI can use it.
 
-===== Write and configure a model reader class
+===== A model reader class your application provides
 Write a Java class that implements the OpenAPI
 link:{microprofile-open-api-javadoc-url}/OASModelReader.html[`org.eclipse.microprofile.openapi.OASModelReader`] interface. Your
 model reader code programmatically adds elements to the internal model that OpenAPI
 builds.
 
+Then set the `mp.openapi.model.reader` configuration property to the fully-qualified name of your model reader class.
+
+===== A filter class your application provides
+Write a Java class that implements the OpenAPI link:{microprofile-open-api-javadoc-url}/OASFilter.html[`org.eclipse.microprofile.openapi.OASFilter`] interface.
+Helidon invokes your filter methods for each element of the in-memory model, allowing your code to modify an element or completely remove it from the model.
+
+Then set the `mp.openapi.filter` configuration property to the fully-qualified name of your filter class.
+
 endif::[]
 
+ifdef::se-flavor[]
 ===== Provide a static OpenAPI file
 Add a static file at `META-INF/openapi.yml`, `META-INF/openapi.yaml`,
 or `META-INF/openapi.json`. Tools such as Swagger let you describe your app's API
 and they then generate an OpenAPI document file which you can include in your application
 so OpenAPI can use it.
 
-===== Write and configure a model reader class
-Write a Java class that implements the OpenAPI
-link:{microprofile-open-api-javadoc-url}/OASModelReader.html[`org.eclipse.microprofile.openapi.OASModelReader`] interface. Your
-model reader code programmatically adds elements to the internal model that OpenAPI
-builds.
-
-Change your application's MP configuration to set `mp.openapi.model.reader` as the
-fully-qualified class name of your class.
-
-===== Write and configure a filter class
-Write a Java class that implements the OpenAPI
-link:{microprofile-open-api-javadoc-url}/OASFilter.html[`org.eclipse.microprofile.openapi.OASFilter`] interface.
-As OpenAPI composes its internal model, it invokes your filter with each
-model element _before_ adding the element to the model. Your filter can
-accept the element as-is, modify it, or suppress it.
-
-Change your application's configuration to set `mp.openapi.filter` as the full-qualified
-class name of your class.
-
+endif::se-flavor[]
 // end::furnish-openapi-info[]
 
 // tag::usage-access-endpoint[]
 === Accessing the REST Endpoint
 Once you add the {flavor-uc} OpenAPI dependency to your
 ifdef::mp-flavor[project,]
-ifdef::se-flavor[project and add code to create the `OpenAPISupport` object to your routing,]
-your application automatically responds to the built-in endpoint --
+ifdef::se-flavor[project and, if you are not using auto-discovery, add code to register the `OpenApiFeature` object with your routing,]
+your application responds to the built-in endpoint --
 `/openapi` -- and it returns the OpenAPI document describing the endpoints
 in your application.
```

Comment:
Once you have added the {flavor-uc} OpenAPI dependency to your ifdef::mp-flavor[project,] and, if you are not using auto-discovery, then you can add code to register the `OpenApiFeature` object with your routing. Your application responds to the built-in endpoint --`/openapi` -- and returns the OpenAPI document describing the endpoints in your application.

Reply:
I've adjusted the wording but a bit differently from your suggestion so it reads well in both the SE and MP contexts with all the conditionals. Please take a look after the next push and make sure it's OK.